src/som/interpreter/actors/AbstractPromiseResolutionNode.java

```diff
@@ -51,15 +63,19 @@ protected void chainPromise(final SResolver resolver,
       final boolean isBreakpointOnPromiseResolution) {
     assert resolver.assertNotCompleted();
     SPromise promiseToBeResolved = resolver.getPromise();
-
+    boolean breakpointOnResolution = isBreakpointOnPromiseResolution;
     synchronized (promiseValue) {
       Resolution state = promiseValue.getResolutionStateUnsync();
       if (SPromise.isCompleted(state)) {
         resolvePromise(state, resolver, promiseValue.getValueUnsync(),
-           isBreakpointOnPromiseResolution);
+            breakpointOnResolution);
       } else {
         synchronized (promiseToBeResolved) { // TODO: is this really deadlock free?
-          promiseToBeResolved.setTriggerResolutionBreakpointOnUnresolvedChainedPromise(isBreakpointOnPromiseResolution);
+          // TODO: check this condition.
```

Comment:
no, apart from the situation I described in the link I don't think there is another case where the flag is false. I just had the doubt why in the case that the resolver is null, the promiseValue is the one that has the flag true? If you consider that the condition is correct then please remove the TODO.